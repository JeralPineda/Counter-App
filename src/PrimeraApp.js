//FC
import React from 'react';
import PropTypes from 'prop-types';

const PrimeraApp = ({ saludo, subtitulo }) => {
   return (
      <>
         {/* <pre>{JSON.stringify(saludo, null, 3)}</pre>0 */}
         <h1>{saludo}</h1>
         <p>{subtitulo}</p>
      </>
   );
};

// Obligamos a que se mande la popiedad saludo
PrimeraApp.propTypes = {
   saludo: PropTypes.string.isRequired,
};

// DefaulProps
PrimeraApp.defaultProps = {
   subtitulo: 'Soy Jeral Pineda',
};

export default PrimeraApp;
