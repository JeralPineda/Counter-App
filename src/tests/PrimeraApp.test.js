import React from 'react';
// import { render } from '@testing-library/react';
import { shallow } from 'enzyme';

import PrimeraApp from '../PrimeraApp';

describe('Prueba en <PrimeraApp />', () => {
   // Con la de crate-react-app
   /**  test('debe de mostrar el mensaje "Holas, soy Jeral"', () => {
      const saludo = 'Hola, soy Jeral';

      //    Renderizar el componente de react
      const { getByText } = render(<PrimeraApp saludo={saludo} />);

      expect(getByText(saludo)).toBeInTheDocument();
   }); **/

   //    Con enzyme
   test('debe de mostrar <PrimeraApp /> correctamente', () => {
      const saludo = 'Hola, soy Jeral';

      const wrapper = shallow(<PrimeraApp saludo={saludo} />);

      expect(wrapper).toMatchSnapshot();
   });

   test('debe de mostrar el subtitulo enviado por props', () => {
      const saludo = 'Hola, soy Jeral';
      const subtitulo = 'Soy Jeral Pineda';

      const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subtitulo} />);

      //   Revisaremos el p√°rrafo que contiene el subtitulo
      const textoParrafo = wrapper.find('p').text();

      expect(textoParrafo).toBe(subtitulo);
   });
});
