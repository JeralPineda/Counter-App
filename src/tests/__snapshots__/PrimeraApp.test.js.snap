// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <PrimeraApp /> debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola, soy Jeral
  </h1>
  <p>
    Soy Jeral Pineda
  </p>
</Fragment>
`;
